session/session_test.go

```diff
@@ -70,6 +118,85 @@ func (s *SessionTestSuite) TestHas() {
 	s.False(s.session.Has("nilKey"))
 }
 
+func (s *SessionTestSuite) TestInvalidate() {
+	oldID := s.session.GetID()
+	s.session.Put("foo", "bar")
+	all := s.session.All()
+	s.Equal("bar", all["foo"])
+
+	s.session.Flash("name", "Krishan")
+	s.True(s.session.Has("name"))
+
+	s.driver.On("Destroy", oldID).Return(nil).Once()
+	s.Nil(s.session.Invalidate())
+
+	s.False(s.session.Has("name"))
+	s.NotEqual(oldID, s.session.GetID())
+	s.Equal(map[string]any{}, s.session.All())
+
+	oldID = s.session.GetID()
+	s.driver.On("Destroy", oldID).Return(errors.New("error")).Once()
+	s.Equal(errors.New("error"), s.session.Invalidate())
+	s.Equal(oldID, s.session.GetID())
+}
+
+func (s *SessionTestSuite) TestKeep() {
+	s.session.Flash("name", "Krishan")
+	s.session.Put("age", 22)
+	s.session.Put("_flash.old", []string{"language"})
+	s.Equal([]string{"name"}, s.session.Get("_flash.new"))
+
+	s.session.Keep("name", "age", "language")
+	s.Equal([]string{"name", "age", "language"}, s.session.Get("_flash.new"))
+	s.Equal([]string{}, s.session.Get("_flash.old"))
+}
+
+func (s *SessionTestSuite) TestMigrate() {
+	oldID := s.session.GetID()
+	s.Nil(s.session.migrate())
+	s.NotEqual(oldID, s.session.GetID())
+
+	oldID = s.session.GetID()
+	s.driver.On("Destroy", oldID).Return(nil).Once()
+	s.Nil(s.session.migrate(true))
+	s.NotEqual(oldID, s.session.GetID())
+}
+
+func (s *SessionTestSuite) TestMissing() {
+	s.session.Put("foo", "bar")
+	s.False(s.session.Missing("foo"))
+	s.session.Put("baz", nil)
+	s.False(s.session.Has("baz"))
+	s.False(s.session.Missing("baz"))
+	s.True(s.session.Missing("bogus"))
+}
+
+func (s *SessionTestSuite) TestNow() {
+	s.session.Now("foo", "bar")
+	s.True(s.session.Has("foo"))
+	s.Equal("bar", s.session.Get("foo"))
+
+	s.session.ageFlashData()
+	s.False(s.session.Has("foo"))
+	s.Nil(s.session.Get("foo"))
+}
+
+func (s *SessionTestSuite) TestOnly() {
+	s.session.
+		Put("foo", "bar").
+		Put("baz", "qux")
+
+	s.Equal(map[string]any{"foo": "bar", "baz": "qux"}, s.session.All())
```

Comment:
The map is random, we can't equal `map` like this, need to compare keys one by one.